pennylane/devices/device_api.py

```diff
@@ -0,0 +1,498 @@
+            * ``result[0][0][0][0]``: the first parameter broadcasting choice
+            * ``result[0][0][0][0][0]``: the first value for an array-valued measurement
+
+            With the exception of quantum script batches, dimensions with only a single component should be eliminated.
+
+            For example:
+
+            With a single script and a single measurement process, execute should return just the
+            measurement value in a numpy array. ``shape`` currently accepts a device, as historically devices
+            stored shot information. In the future, this method will accept an ``ExecutionConfig`` instead.
+
+            >>> qs = qml.tape.QuantumTape(measurements=qml.expval(qml.PauliZ(0))])
+            >>> qs.shape(dev)
+            ()
+            >>> dev.execute(qs)
+            array(1.0)
+
+            If execute recieves a batch of scripts, then it should return a tuple of results:
+
+            >>> dev.execute([qs, qs])
+            (array(1.0), array(1.0))
+            >>> dev.execute([qs])
+            (array(1.0),)
+
+            If the script has multiple measurments, then the device should return a tuple of measurements.
+
+            >>> qs = qml.tape.QuantumTape(measurements=[qml.expval(qml.PauliZ(0)), qml.probs(wires=(0,1))])
+            >>> qs.shape(dev)
+            ((), (4,))
+            >>> dev.execute(qs)
+            (array(1.0), array([1., 0., 0., 0.]))
+
+        """
+        raise NotImplementedError
+
+    @classmethod
+    def supports_derivatives(
+        cls, execution_config: ExecutionConfig = DefaultExecutionConfig
+    ) -> bool:
+        """Determine whether or not a gradient is available with a given execution configuration.
+
+        Default behaviour assumes first order derivatives exist if :meth:`~.differentiate` is overriden.
+
+        Args:
+            execution_config (ExecutionConfig): A description of the hyperparameters for the desired computation.
+              Note that this method accepts only a single ``ExecutionConfig``, not an iterable of them.
+
+        Returns:
+            Bool
+
+        **Example:**
+
+        For example, the Python device will support device differentiation via the adjoint differentiation algorithm
+        if the order is ``1`` and the execution occurs with no shots ``shots=None``.
+
+        >>> config = ExecutionConfig(derivative_order=1, shots=None, gradient_method="device")
+        >>> dev.supports_differentiation(config)
+        True
+        >>> config = ExecutionConfig(derivative_order=1, shots=10, gradient_method="device")
+        >>> dev.supports_differentiation(config)
+        False
+        >>> config = ExecutionConfig(derivative_order=2, shots=None, gradient_method="device")
+        >>> dev.supports_differentiation(config)
+        False
+
+        **Backpropagation:**
+
+        This method is also used be to validate support for backpropagation derivatives. Backpropagation
```

Comment:
It is not the case at the moment, right?

```
       if execution_config.gradient_method != "device":
            return False
```

Reply:
For `DefualtQubit2`, we will override this and define:
```
if execution_config.gradient_method == "backprop":
    return True
```